lib/std/testing.zig

```diff
@@ -11,7 +11,7 @@ pub var allocator_instance = LeakCountAllocator.init(&base_allocator_instance.al
 pub const failing_allocator = &failing_allocator_instance.allocator;
 pub var failing_allocator_instance = FailingAllocator.init(&base_allocator_instance.allocator, 0);
 
-pub var base_allocator_instance = std.heap.ThreadSafeFixedBufferAllocator.init(allocator_mem[0..]);
+pub var base_allocator_instance = std.mem.sanityWrap(std.heap.ThreadSafeFixedBufferAllocator.init(allocator_mem[0..]));
```

Comment:
The doc comments for the allocator say 

>  /// Detects and asserts if the std.mem.Allocator interface is violated 

Which looks to me like it's for testing *implementations* of the Allocator interface, rather than testing the usage of the Allocator high level API functions. In this case I don't think it makes sense to put the wrapper here, since for testing purposes we are not trying to test if the LeakCountAllocator code is correct, we are trying to test if the user code  that is using the Allocator provided by the LeakCountAllocator is correct.

Reply:
`ValidationAllocator` validates that the allocator functions are called with valid parameters, and that the underlying allocator returns a valid value.

i.e. for `alloc`, `ValidationAllocator ensures `len > 0`, `ptr_align` is valid, that `len` is aligned by `len_align` and `len >= len_align`.  Once it returns, it validates that the `len` compiles with the requested `len_align` and that the pointer is aligned by `ptr_align`.